open_spiel/games/lewis_signaling.cc

```diff
@@ -135,12 +135,16 @@ void LewisSignalingState::ObservationTensor(Player player,
   offset += 1;
 
   // one-hot vector for the state/message
-  if (static_cast<Players>(cur_player_) == Players::kSender) {
-    (*values)[offset + state_] = 1;
-    offset += num_states_;
-  } else if (static_cast<Players>(cur_player_) == Players::kReceiver) {
-    (*values)[offset + message_] = 1;
-    offset += num_messages_;
+  if (static_cast<Players>(player) == Players::kSender) {
+    if (state_ != -1) {
```

Comment:
only in the last step state_ == -1, is that right?

Reply:
State is -1 before the chance node selects it. So it is just an additional check that the state has been assigned.

Reply:
Hi @vbhatt-cs , can you change the -1 to `pyspiel.PlayerId.CHANCE` ?

Reply:
Oh wait this is in C++, nevermind. Maybe just add a comment to explain this?

Reply:
Would making a new const variable and doing `state_ != kUnassignedValue` work?

Reply:
Yes, would be great (keep it in the Lewis signaling game namespace)